node/silkworm/stagedsync/stage_hashstate.cpp

```diff
@@ -350,6 +360,7 @@ StageResult HashState::hash_from_plaincode(db::RWTxn& txn) {
         if (!is_stopping()) {
             if (!collector_->empty()) {
                 source = db::open_cursor(*txn, db::table::kHashedCodeHash);
+                std::unique_lock lck(log_mtx_);
```

Comment:
`collector_->load` could be a long operation, better perform it outside the lock.